stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/experimental/plugin/adapter/ChatClientStateCalls.kt

```diff
@@ -6,38 +6,46 @@ import io.getstream.chat.android.client.api.models.QueryChannelsRequest
 import io.getstream.chat.android.core.internal.InternalStreamChatApi
 import io.getstream.chat.android.offline.ChatDomain
-import io.getstream.chat.android.offline.experimental.channel.QueryChannelReference
-import io.getstream.chat.android.offline.experimental.channel.thread.RepliesQueryReference
-import io.getstream.chat.android.offline.experimental.querychannels.QueryChannelsReference
+import io.getstream.chat.android.offline.experimental.channel.state.ChannelState
+import io.getstream.chat.android.offline.experimental.channel.thread.state.ThreadState
+import io.getstream.chat.android.offline.experimental.plugin.state.StateRegistry
+import io.getstream.chat.android.offline.experimental.querychannels.state.QueryChannelsState
 import io.getstream.chat.android.offline.request.QueryChannelPaginationRequest
 
 @InternalStreamChatApi
 /**
  * Adapter for [ChatClient] that wraps some of it's request with [io.getstream.chat.android.offline.experimental.plugin.QueryReference].
  */
-public class ChatClientReferenceAdapter(private val chatClient: ChatClient) {
+internal class ChatClientStateCalls(private val chatClient: ChatClient, private val state: StateRegistry) {
     /** Reference request of the channels query. */
-    public fun queryChannels(request: QueryChannelsRequest): QueryChannelsReference =
-        QueryChannelsReference(request, chatClient)
+    internal fun queryChannels(request: QueryChannelsRequest): QueryChannelsState {
+        chatClient.queryChannels(request).enqueue()
+        return state.queryChannels(request.filter, request.querySort)
+    }
 
     /** Reference request of the channel query. */
-    public fun queryChannel(
+    private fun queryChannel(
         channelType: String,
         channelId: String,
         request: QueryChannelRequest,
-    ): QueryChannelReference = QueryChannelReference(channelType, channelId, request, chatClient)
+    ): ChannelState {
+        chatClient.queryChannel(channelType, channelId, request).enqueue()
+        return state.channel(channelType, channelId)
+    }
 
     /** Reference request of the watch channel query. */
-    public fun watchChannel(cid: String, limit: Int = DEFAULT_MESSAGE_LIMIT): QueryChannelReference {
+    internal fun watchChannel(cid: String, limit: Int = DEFAULT_MESSAGE_LIMIT): ChannelState {
         val (channelType, channelId) = cid.cidToTypeAndId()
         val userPresence = runCatching { ChatDomain.instance().userPresence }.getOrDefault(false)
         val request = QueryChannelPaginationRequest(limit).toWatchChannelRequest(userPresence)
         return queryChannel(channelType, channelId, request)
     }
 
     /** Reference request of the get thread replies query. */
-    public fun getReplies(messageId: String, limit: Int = DEFAULT_MESSAGE_LIMIT): RepliesQueryReference =
-        RepliesQueryReference(messageId, limit, chatClient)
+    internal fun getReplies(messageId: String, limit: Int = DEFAULT_MESSAGE_LIMIT): ThreadState {
```

Comment:
Does it make sense to add one more parameter `scope` and then call:
```
scope.launch{
chatClient.getReplies(messageId, limit).await()
}
```
instead? This way calls could be easily cancelled 🤔

Reply:
Thinking about it, it does make sense, because simply class Call.enqueue will cause memory leaks, because the calls won't be canceled =|. But how customers using Java will use this API?

Reply:
I think in this case enqueue does the same thing - we could have a scope to potentially launch a coroutine, but not much would be gained if this starts an async/parallel request anyway.